diagnostics/profile.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2024 The Erigon Authors
+// This file is part of Erigon.
+//
+// Erigon is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Erigon is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with Erigon. If not, see <http://www.gnu.org/licenses/>.
+
+package diagnostics
+
+import (
+	"fmt"
+	"net/http"
+	"runtime/pprof"
+
+	diaglib "github.com/erigontech/erigon-lib/diagnostics"
+)
+
+func SetupProfileAccess(metricsMux *http.ServeMux, diag *diaglib.DiagnosticClient) {
+	if metricsMux == nil {
+		return
+	}
+
+	//handle all pprof, supported: goroutine, threadcreate, heap, allocs, block, mutex
+	metricsMux.HandleFunc("/pprof/", func(w http.ResponseWriter, r *http.Request) {
+		profile := r.URL.Path[len("/pprof/"):]
```

Comment:
You can use [strings.TrimPrefix](https://pkg.go.dev/strings#TrimPrefix). Alternatively, you can also use HTTP params using `r.URL.Query().Get("profile")`, but this changes your request format to `/pprof?profile=<name>`.

Reply:
done